ome/resource_template.go

```diff
@@ -458,19 +461,43 @@ func (r resourceTemplate) Create(ctx context.Context, req tfsdk.CreateResourceRe
 	}
 	//PropogateVlan is default true and is not available in the response from OME,hence setting it to true here to persist in state
 	omeVlan.PropagateVLAN = true
-	template.RefdeviceServicetag.Value = plan.RefdeviceServicetag.Value
-	template.ReftemplateName.Value = plan.ReftemplateName.Value
-	template.RefdeviceID.Value = plan.RefdeviceID.Value
-	if plan.ReftemplateName.Value != "" {
-		template.RefdeviceID.Value = omeTemplateData.SourceDeviceID
-	}
-	template.Content.Value = plan.Content.Value
-	template.ViewType.Value = plan.ViewType.Value
-	template.DeviceType.Value = plan.DeviceType.Value
-	template.FQDDS.Value = plan.FQDDS.Value // The default value of fqdds is set to `All`. So if the config doesn't have any value specified, the default value in the plan is `All`.
-	template.JobRetryCount = plan.JobRetryCount
-	template.SleepInterval = plan.SleepInterval
-	template.IdentityPoolName.Value = plan.IdentityPoolName.Value
+	if !plan.RefdeviceServicetag.IsUnknown() {
+		template.RefdeviceServicetag = plan.RefdeviceServicetag
```

Comment:
same condition here. I would love to see why we are checking these unknowns for each and every attribute because in the older version we're not doing that.

Reply:
we have tried to unset the unknown check for the template resource and encountered invalid object error
  /root/priyanshu/OME/terraform-provider-ome/ome/resource_template_test.go:271: Step 1/1 error: Error running apply: exit status 1
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citdtest.identity_pool_name. All values must be known after
        apply, so this is always a bug in the provider and should be reported in the
        provider's own repository. Terraform will still save the other known object
        values in the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citdtest.refdevice_id. All values must be known after apply, so
        this is always a bug in the provider and should be reported in the provider's
        own repository. Terraform will still save the other known object values in
        the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citdtest.refdevice_servicetag. All values must be known after
        apply, so this is always a bug in the provider and should be reported in the
        provider's own repository. Terraform will still save the other known object
        values in the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citdtest.reftemplate_name. All values must be known after apply,
        so this is always a bug in the provider and should be reported in the
        provider's own repository. Terraform will still save the other known object
        values in the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citctest.identity_pool_name. All values must be known after
        apply, so this is always a bug in the provider and should be reported in the
        provider's own repository. Terraform will still save the other known object
        values in the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citctest.refdevice_id. All values must be known after apply, so
        this is always a bug in the provider and should be reported in the provider's
        own repository. Terraform will still save the other known object values in
        the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citctest.refdevice_servicetag. All values must be known after
        apply, so this is always a bug in the provider and should be reported in the
        provider's own repository. Terraform will still save the other known object
        values in the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citctest.reftemplate_name. All values must be known after apply,
        so this is always a bug in the provider and should be reported in the
        provider's own repository. Terraform will still save the other known object
        values in the state.
--- FAIL: TestTemplateCreation_CreateImportTemplate (4.89s)
FAIL
FAIL    terraform-provider-ome/ome      298.035s

Reply:
tried to unset the unknown check in template resource
we have encountered the error while unsetting it 
`  /root/priyanshu/OME/terraform-provider-ome/ome/resource_template_test.go:271: Step 1/1 error: Error running apply: exit status 1
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citdtest.identity_pool_name. All values must be known after
        apply, so this is always a bug in the provider and should be reported in the
        provider's own repository. Terraform will still save the other known object
        values in the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citdtest.refdevice_id. All values must be known after apply, so
        this is always a bug in the provider and should be reported in the provider's
        own repository. Terraform will still save the other known object values in
        the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citdtest.refdevice_servicetag. All values must be known after
        apply, so this is always a bug in the provider and should be reported in the
        provider's own repository. Terraform will still save the other known object
        values in the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citdtest.reftemplate_name. All values must be known after apply,
        so this is always a bug in the provider and should be reported in the
        provider's own repository. Terraform will still save the other known object
        values in the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citctest.identity_pool_name. All values must be known after
        apply, so this is always a bug in the provider and should be reported in the
        provider's own repository. Terraform will still save the other known object
        values in the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citctest.refdevice_id. All values must be known after apply, so
        this is always a bug in the provider and should be reported in the provider's
        own repository. Terraform will still save the other known object values in
        the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citctest.refdevice_servicetag. All values must be known after
        apply, so this is always a bug in the provider and should be reported in the
        provider's own repository. Terraform will still save the other known object
        values in the state.
        Error: Provider returned invalid result object after apply
        After the apply operation, the provider still indicated an unknown value for
        ome_template.citctest.reftemplate_name. All values must be known after apply,
        so this is always a bug in the provider and should be reported in the
        provider's own repository. Terraform will still save the other known object
        values in the state.
--- FAIL: TestTemplateCreation_CreateImportTemplate (4.89s)
FAIL
FAIL    terraform-provider-ome/ome      298.035s`

Reply:
template acceptance test ran successfully with unknown checks.